provider/server.go

```diff
@@ -105,35 +224,197 @@ func (s *RawProviderServer) UpgradeResourceState(ctx context.Context, req *tfplu
 func (s *RawProviderServer) Configure(ctx context.Context, req *tfplugin5.Configure_Request) (*tfplugin5.Configure_Response, error) {
 	response := &tfplugin5.Configure_Response{}
 
-	providerConfig, err := msgpack.Unmarshal(req.Config.Msgpack, GetConfigObjectType())
+	providerConfig, err := msgpack.Unmarshal(req.Config.Msgpack, getConfigObjectType())
 	if err != nil {
 		return response, err
 	}
 
-	configFile := providerConfig.GetAttr("config_file")
-	var kubeconfig string
+	overrides := &clientcmd.ConfigOverrides{}
+	loader := &clientcmd.ClientConfigLoadingRules{}
 
-	// if no config specified, try the known default locations
-	if configFile.IsNull() || configFile.AsString() == "" {
-		h := os.Getenv("HOME")
-		if h == "" {
-			h = os.Getenv("USERPROFILE") // windows
+	var configPath cty.Value
+	if configPathEnv, ok := os.LookupEnv("KUBE_CONFIG_PATH"); ok && configPathEnv != "" {
```

Comment:
I'm still a bit confused as to why we're not supporting `KUBECONFIG` but we are reimplementing it as `KUBE_CONFIG_PATH`. Like what's the use-case you foresee for this? Given that all these `KUBE_` env vars are not standard to kubectl couldn't the user just supply variables to the provider block and use `TF_` env vars if they want to be able to configure the provider through the environment?

Reply:
KUBECONFIG has a high chance of being inadvertently picked up and loaded "under the radar" and issue history has shown this catches some users on the wrong foot. I decided to experiment with being completely explicit and requiring users to deliberately opt into loading it. I'm looking to see if this is a better UX in the end. We can always reverse to being implicit if there is demand for it.